nilearn/interfaces/tests/test_bids.py

```diff
@@ -12,13 +13,188 @@
+
+    bids_path = create_fake_bids_dataset(base_dir=tmp_path,
+                                            n_sub=1,
+                                            n_ses=1,
+                                            tasks=['main'],
+                                            n_runs=[1])
+
+    StartTime = _infer_slice_timing_start_time_from_dataset(
+        bids_path = tmp_path / bids_path / "derivatives",
+        filters=[('task', 'main')]
+        )
+    assert StartTime is None
+
+    add_metadata_to_bids_dataset(
+        bids_path = tmp_path / bids_path,
+        metadata = {"StartTime": 1.0})
+    StartTime = _infer_slice_timing_start_time_from_dataset(
+        bids_path = tmp_path / bids_path / "derivatives",
+        filters=[('task', 'main')]
+        )
+    assert StartTime == 1.0    
+
+
+def _rm_all_json_files_from_bids_dataset(bids_path):
+    """Remove all json and make sure that get_bids_files does not find any."""
+    [x.unlink() for x in bids_path.glob("**/*.json")]
+    selection = get_bids_files(bids_path, file_type='json', sub_folder=True)
+    assert selection == []
+    selection = get_bids_files(bids_path, file_type='json', sub_folder=False)
+    assert selection == []  
+
+
+def test_get_bids_files_inheritance_principle_root_folder(tmp_path):
+    """Check if json files are found if in root folder of a dataset.
+
+    see https://bids-specification.readthedocs.io/en/latest/common-principles.html#the-inheritance-principle #  noqa
+    """
+    
+
+    bids_path = create_fake_bids_dataset(base_dir=tmp_path,
+                                         n_sub=1,
+                                         n_ses=1,
+                                         tasks=['main'],
+                                         n_runs=[1])
+    
+    _rm_all_json_files_from_bids_dataset(bids_path) 
+
+    # add json file to root of dataset
+    json_file='task-main_bold.json'
+    json_file = add_metadata_to_bids_dataset(
+        bids_path=bids_path,
+        metadata={"RepetitionTime": 1.5},
+        json_file=json_file
+    )
+    assert json_file.exists()
+
+    # make sure that get_bids_files finds the json file 
+    # but only when looking in root of dataset
+    selection = get_bids_files(bids_path,
+                               file_tag="bold",
+                               file_type='json',
+                               filters = [ ('task', 'main')],
+                               sub_folder=True)
+    assert selection == []
+    selection = get_bids_files(bids_path,
+                               file_tag="bold",
+                               file_type='json',
+                               filters = [ ('task', 'main')],
+                               sub_folder=False)
+    assert selection != []
+    assert selection[0] == str(json_file)
+
+
+@pytest.mark.xfail(
+        reason=("get_bids_files does not find json files"
+                " that are directly in the subject folder of a dataset.")
+        )
```

Comment:
I have not seen xfail used in the test suite so not sure if there is any guideline about using them. 

https://docs.pytest.org/en/7.1.x/how-to/skipping.html#xfail-mark-test-functions-as-expected-to-fail

Will make a strict xfail so that an XPASS (“unexpectedly passing”) results from this test will fail the test suite.